src/sentry/api/serializers/models/incident.py

```diff
@@ -19,9 +19,15 @@ def get_attrs(self, item_list, user, **kwargs):
         ).select_related("project"):
             incident_projects[incident_project.incident_id].append(incident_project.project.slug)
 
+        alert_rules = {
```

Comment:
You should also include `attach_foreignkey(item_list, Incident.alert_rule, related=("snuba_query",))` (could remove it from the detailed serializer too).

This is similar to `Incident.objects.filter(...).select_related('alert_rule')`, except that it works on a list of models rather than a queryset. This makes sure we fetch all the alert rules in a single query.